modules/apps/object/object-service/src/main/java/com/liferay/object/internal/notification/term/contributor/ObjectDefinitionNotificationTermEvaluator.java

```diff
@@ -55,58 +65,179 @@ public String evaluate(Context context, Object object, String termName)
 
 		Map<String, Object> termValues = (Map<String, Object>)object;
 
-		if (termName.contains("_CREATOR")) {
+		for (UnsafeTriFunction
+				<Context, String, Map<String, Object>, String, PortalException>
+					evaluatorFunction : _evaluatorsFunctions) {
+
+			String termValue = evaluatorFunction.apply(
+				context, termName, termValues);
+
+			if (termValue != null) {
+				return termValue;
+			}
+		}
+
+		return termName;
+	}
+
+	private String _evaluateAuthor(
+			Context context, String termName, Map<String, Object> termValues)
+		throws PortalException {
+
+		String prefix = StringUtil.toUpperCase(
+			_objectDefinition.getShortName());
+
+		if (!termName.equals("[%" + prefix + "_CREATOR%]") &&
+			!termName.equals("[%" + prefix + "_AUTHOR_EMAIL_ADDRESS%]") &&
+			!termName.equals("[%" + prefix + "_AUTHOR_FIRST_NAME%]") &&
+			!termName.equals("[%" + prefix + "_AUTHOR_ID%]") &&
+			!termName.equals("[%" + prefix + "_AUTHOR_LAST_NAME%]") &&
+			!termName.equals("[%" + prefix + "_AUTHOR_MIDDLE_NAME%]") &&
+			!termName.equals("[%" + prefix + "_AUTHOR_PREFIX%]") &&
+			!termName.equals("[%" + prefix + "_AUTHOR_SUFFIX%]")) {
+
+			return null;
+		}
+
+		User user = _userLocalService.getUser(
+			GetterUtil.getLong(termValues.get("creator")));
+
+		if (!FeatureFlagManagerUtil.isEnabled("LPS-171625")) {
 			if (context.equals(Context.RECIPIENT)) {
 				return String.valueOf(termValues.get("creator"));
 			}
 
-			User user = _userLocalService.getUser(
-				GetterUtil.getLong(termValues.get("creator")));
-
 			return user.getFullName(true, true);
 		}
 
-		Map<String, Long> objectFieldIds =
-			_objectFieldIdsDCLSingleton.getSingleton(
-				this::_createObjectFieldIds);
+		if (user == null) {
+			return null;
+		}
 
-		ObjectField objectField = _objectFieldLocalService.fetchObjectField(
-			objectFieldIds.get(termName));
+		String suffix = StringUtil.removeSubstring(
+			termName, "[%" + prefix + "_AUTHOR_");
 
-		if (objectField == null) {
-			return termName;
+		if (StringUtil.equals(termName, suffix)) {
+			suffix = StringUtil.removeSubstring(termName, "[%" + prefix + "_");
```

Comment:
remove it because we are going to handle it above, so:

````java
_getTermValue(context, StringUtil.removeSubstring(termName, "[%" + prefix + "_AUTHOR_"), user);
````